### PR TITLE
Added tests for Detector nuget package

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: build/testDetector.sh [hunk 1]
```diff
@@ -11,15 +11,42 @@ declare -r REPO_DIR=$( cd $( dirname "$0" ) && cd .. && pwd )
 # Load all variables
 source $REPO_DIR/build/__variables.sh
 
+# A hint to Detector.csproj file to generate the nuget package in a specific way so that
```
Comment (kichalla) [inline build/testDetector.sh:14]: Will explain in meeting as to why we are doing this

File: src/Detector/Detector.csproj [hunk 2]
```diff
@@ -35,18 +35,26 @@
     <PackageProjectUrl>https://github.com/microsoft/oryx</PackageProjectUrl>
     <PackageTags>Oryx Detector OryxDetector</PackageTags>
     <Description>A library to detect tools and frameworks used in an application.</Description>
+    <Version>1.0.0-pre-</Version>
```
Comment (gpcastro) [inline src/Detector/Detector.csproj:38]: can we just have latest here?

File: tests/Oryx.Tests.Common/TestTempDirTestFixture.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 using System;
 using System.IO;
 
-namespace Microsoft.Oryx.Tests.Common
+namespace Microsoft.Oryx.Detector.Tests
```
Comment (gpcastro) [inline tests/Oryx.Tests.Common/TestTempDirTestFixture.cs:9]: why the rename?

Comment (kichalla) [reply tests/Oryx.Tests.Common/TestTempDirTestFixture.cs:9]: This was by mistake. I will revert it actually.

File: src/Detector/Detector.csproj (hunk 2 above)
Comment (kichalla) [reply src/Detector/Detector.csproj:38]: Prerelease packages on NuGet.org typically start with `pre`, so I am using that convention here.

**Pull request merged**